sql/catalog/lease: TestLeaseAtLatestVersion failed

sql/catalog/lease.TestLeaseAtLatestVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16398043?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16398043?buildTab=artifacts#/) on master @ [22c9923eb9854b9e31068234599edce7c856fbde](https://github.com/cockroachdb/cockroach/commits/22c9923eb9854b9e31068234599edce7c856fbde):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestLeaseAtLatestVersion (14m35s)

```
Stack: 

```
goroutine 74736 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLeaseAtLatestVersion
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/5632b52c1c67b61307e1d8a1ca2dd18d/logTestLeaseAtLatestVersion373407812
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseAtLatestVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41153

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -1140,7 +1140,7 @@ func TestLeaseAtLatestVersion(t *testing.T) {
 				LeaseAcquiredEvent: func(desc catalog.Descriptor, _ error) {
 					if desc.GetName() == "kv" {
 						var err error
-						if desc.GetVersion() != 2 {
+						if desc.GetVersion() < 2 {
 							err = errors.Errorf("not seeing latest version")
 						}
 						errChan <- err

